csharp_dotnetcore/Using-Cards/Using_Cards/CardsBot.cs

```diff
@@ -0,0 +1,339 @@
+            // So we need to create a list of attachments on the activity.
+            reply.Attachments = new List<Attachment>();
+
+            // Decide which type of card(s) we are going to show the user
+            switch (text)
+            {
+                case "herocard":
+                case "hero":
+                    reply.Attachments.Add(GetHeroCard().ToAttachment());
+                    break;
+                case "thumbnailcard":
+                case "thumbnail":
+                    reply.Attachments.Add(GetThumbnailCard().ToAttachment());
+                    break;
+                case "receiptcard":
+                case "receipt":
+                    reply.Attachments.Add(GetReceiptCard().ToAttachment());
+                    break;
+                case "signincard":
+                case "signin":
+                case "sign":
+                    reply.Attachments.Add(GetSigninCard().ToAttachment());
+                    break;
+                case "animationcard":
+                case "animation":
+                    reply.Attachments.Add(GetAnimationCard().ToAttachment());
+                    break;
+                case "videocard":
+                case "video":
+                    reply.Attachments.Add(GetVideoCard().ToAttachment());
+                    break;
+                case "audiocard":
+                case "audio":
+                    reply.Attachments.Add(GetAudioCard().ToAttachment());
+                    break;
+                default:
+                    // Send all cards in a carousel if the user wants to see all cards.
+                    reply.AttachmentLayout = AttachmentLayoutTypes.Carousel;
+                    reply.Attachments.Add(GetHeroCard().ToAttachment());
+                    reply.Attachments.Add(GetThumbnailCard().ToAttachment());
+                    reply.Attachments.Add(GetReceiptCard().ToAttachment());
+                    reply.Attachments.Add(GetSigninCard().ToAttachment());
+                    reply.Attachments.Add(GetAnimationCard().ToAttachment());
+                    reply.Attachments.Add(GetVideoCard().ToAttachment());
+                    reply.Attachments.Add(GetAudioCard().ToAttachment());
+                    break;
+
+            }
+
+            // Send the card(s) to the user as an attachment to the activity
+            await dc.Context.SendActivityAsync(reply, cancellationToken);
+
+            // Give the user instructions about what to do next
+            await dc.Context.SendActivityAsync("Type anything to see another card.", cancellationToken: cancellationToken);
+
+            return await dc.EndAsync(cancellationToken: cancellationToken);
+        }
+        private static HeroCard GetHeroCard()
+        {
+            var heroCard = new HeroCard
+            {
+                Title = "BotFramework Hero Card",
+                Subtitle = "Your bots — wherever your users are talking",
```

Comment:
I don't get "Your bots". Consider checking with Vishwac and updating the subtitles.